diff --git a/browser/base/content/browser-init.js b/browser/base/content/browser-init.js
index cca7b67..ee0d1a9 100644
--- a/browser/base/content/browser-init.js
+++ b/browser/base/content/browser-init.js
@@ -237,6 +237,24 @@ var gBrowserInit = {
     // correctly.
     window.TabBarVisibility.update();
 
+    // If a list of languages were passed, use them.
+    let rotundaLocale =
+      ChromeUtils.rotundaGetString("locale.all") ||
+      ChromeUtils.rotundaGetString("navigator.language");
+    // If locale.all was NOT passed, but locale.language and locale.region was,
+    // fall back to it instead.
+    if (!rotundaLocale) {
+      let language = ChromeUtils.rotundaGetString("locale.language");
+      let region = ChromeUtils.rotundaGetString("locale.region");
+      if (language && region) {
+        rotundaLocale = language + "-" + region + ", " + language;
+      }
+    }
+    // Set the locale if it was found.
+    if (rotundaLocale) {
+      Services.prefs.setCharPref("intl.accept_languages", rotundaLocale);
+    }
+
     if (
       Services.prefs.getBoolPref(
         "toolkit.legacyUserProfileCustomizations.windowIcon",
diff --git a/intl/components/moz.build b/intl/components/moz.build
index a3319c7..6594778 100644
--- a/intl/components/moz.build
+++ b/intl/components/moz.build
@@ -72,6 +72,7 @@ if not CONFIG["JS_STANDALONE"]:
 
 LOCAL_INCLUDES += [
     "/intl/icu_capi/bindings/cpp",
+    "/rotundacfg",
 ]
 
 if not CONFIG["MOZ_SYSTEM_ICU"]:
diff --git a/intl/components/src/Locale.cpp b/intl/components/src/Locale.cpp
index 718d881..cda3fd9 100644
--- a/intl/components/src/Locale.cpp
+++ b/intl/components/src/Locale.cpp
@@ -20,11 +20,78 @@
 
 #include "unicode/uloc.h"
 #include "unicode/utypes.h"
+#include "MaskConfig.hpp"
 
 namespace mozilla::intl {
 
 using namespace intl::LanguageTagLimits;
 
+
+// Helper methods for header file
+
+const char* Locale::GetRotundaLanguage() {
+  static std::string language;
+  if (auto locale = MaskConfig::Profile().getLocale()) {
+    if (auto value = locale->getLanguage()) {
+      language = *value;
+      return language.c_str();
+    }
+  }
+  return nullptr;
+}
+
+const char* Locale::GetRotundaRegion() {
+  static std::string region;
+  if (auto locale = MaskConfig::Profile().getLocale()) {
+    if (auto value = locale->getRegion()) {
+      region = *value;
+      return region.c_str();
+    }
+  }
+  return nullptr;
+}
+
+const char* Locale::GetRotundaScript() {
+  static std::string script;
+  if (auto locale = MaskConfig::Profile().getLocale()) {
+    if (auto value = locale->getScript()) {
+      script = *value;
+      return script.c_str();
+    }
+  }
+  return nullptr;
+}
+
+// Publicly exposed methods
+
+const char* Locale::GetRotundaLocale() {
+  static std::string locale;
+
+  if (auto localeProfile = MaskConfig::Profile().getLocale()) {
+    if (auto lang = localeProfile->getLanguage()) {
+      if (auto region = localeProfile->getRegion()) {
+        locale = (*lang + "-" + *region);
+      }
+    }
+  }
+  if (auto navigator = MaskConfig::Profile().getNavigator()) {
+    if (auto value = navigator->getLanguage()) {
+      locale = *value;
+    }
+  }
+  if (locale.empty())
+    return nullptr;
+  return locale.c_str();
+}
+
+const char* Locale::GetDefaultLocale() {
+  // In Rotunda, GetDefaultLocale is defined outside the header
+  // to access MaskConfig.
+  if (auto value = GetRotundaLocale())
+    return value;
+  return uloc_getDefault();
+}
+
 template <typename CharT>
 bool IsStructurallyValidLanguageTag(Span<const CharT> aLanguage) {
   // unicode_language_subtag = alpha{2,3} | alpha{5,8};
diff --git a/intl/components/src/Locale.h b/intl/components/src/Locale.h
index b7314ec..5426531 100644
--- a/intl/components/src/Locale.h
+++ b/intl/components/src/Locale.h
@@ -208,8 +208,11 @@ class MOZ_STACK_CLASS Locale final {
 
  private:
   LanguageSubtag mLanguage = {};
+  mutable LanguageSubtag mRotundaLanguage = {};
   ScriptSubtag mScript = {};
+  mutable ScriptSubtag mRotundaScript = {};
   RegionSubtag mRegion = {};
+  mutable RegionSubtag mRotundaRegion = {};
 
   VariantsVector mVariants;
   ExtensionsVector mExtensions;
@@ -338,9 +341,28 @@ class MOZ_STACK_CLASS Locale final {
     }
   };
 
-  const LanguageSubtag& Language() const { return mLanguage; }
-  const ScriptSubtag& Script() const { return mScript; }
-  const RegionSubtag& Region() const { return mRegion; }
+  const LanguageSubtag& Language() const {
+    if (const char* lang = GetRotundaLanguage()) {
+      mRotundaLanguage.Set(mozilla::Span<const char>(lang, strlen(lang)));
+      return mRotundaLanguage;
+    }
+    return mLanguage;
+  }
+  const ScriptSubtag& Script() const {
+    if (const char* script = GetRotundaScript()) {
+      mRotundaScript.Set(mozilla::Span<const char>(script, strlen(script)));
+      return mRotundaScript;
+    }
+    return mScript;
+  }
+  const RegionSubtag& Region() const {
+    if (const char* region = GetRotundaRegion()) {
+      mRotundaRegion.Set(mozilla::Span<const char>(region, strlen(region)));
+      return mRotundaRegion;
+    }
+    return mRegion;
+  }
+
   auto Variants() const { return SubtagEnumeration(mVariants); }
   auto Extensions() const { return SubtagEnumeration(mExtensions); }
   Maybe<Span<const char>> PrivateUse() const {
@@ -518,7 +540,15 @@ class MOZ_STACK_CLASS Locale final {
    *
    * Also see <https://unicode-org.github.io/icu/userguide/locale>.
    */
-  static const char* GetDefaultLocale() { return uloc_getDefault(); }
+  static const char* GetDefaultLocale();
+
+  static const char* GetRotundaLocale();
+
+  static const char* GetRotundaLanguage();
+
+  static const char* GetRotundaRegion();
+
+  static const char* GetRotundaScript();
 
   /**
    * Returns an iterator over all supported locales.
diff --git a/intl/locale/OSPreferences.cpp b/intl/locale/OSPreferences.cpp
index b185d97..b9e80f2 100644
--- a/intl/locale/OSPreferences.cpp
+++ b/intl/locale/OSPreferences.cpp
@@ -410,6 +410,10 @@ bool OSPreferences::GetDateTimeConnectorPattern(const nsACString& aLocale,
  */
 NS_IMETHODIMP
 OSPreferences::GetSystemLocales(nsTArray<nsCString>& aRetVal) {
+  if (const char* rotundaLocale = mozilla::intl::Locale::GetRotundaLocale()) {
+    aRetVal.AppendElement(rotundaLocale);
+    return NS_OK;
+  }
   if (!mSystemLocales.IsEmpty()) {
     aRetVal = mSystemLocales.Clone();
     return NS_OK;
@@ -429,7 +433,10 @@ OSPreferences::GetSystemLocales(nsTArray<nsCString>& aRetVal) {
 
 NS_IMETHODIMP
 OSPreferences::GetSystemLocale(nsACString& aRetVal) {
-  if (!mSystemLocales.IsEmpty()) {
+  if (const char* rotundaLocale = mozilla::intl::Locale::GetRotundaLocale()) {
+    aRetVal = rotundaLocale;
+    return NS_OK;
+  } else if (!mSystemLocales.IsEmpty()) {
     aRetVal = mSystemLocales[0];
   } else {
     AutoTArray<nsCString, 10> locales;
@@ -443,6 +450,10 @@ OSPreferences::GetSystemLocale(nsACString& aRetVal) {
 
 NS_IMETHODIMP
 OSPreferences::GetRegionalPrefsLocales(nsTArray<nsCString>& aRetVal) {
+  if (const char* rotundaLocale = mozilla::intl::Locale::GetRotundaLocale()) {
+    aRetVal.AppendElement(rotundaLocale);
+    return NS_OK;
+  }
   if (!mRegionalPrefsLocales.IsEmpty()) {
     aRetVal = mRegionalPrefsLocales.Clone();
     return NS_OK;
