diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -22,11 +22,6 @@ const { Multilingual } = ChromeUtils.importESModule(
   { global: "current" }
 );
 
-const { DefaultBrowserHelper } = ChromeUtils.importESModule(
-  "chrome://browser/content/preferences/DefaultBrowserHelper.mjs",
-  { global: "current" }
-);
-
 ChromeUtils.defineESModuleGetters(this, {
   BackgroundUpdate: "resource://gre/modules/BackgroundUpdate.sys.mjs",
   UpdateListener: "resource://gre/modules/UpdateListener.sys.mjs",
@@ -374,61 +369,0 @@ Preferences.addSetting({
-Preferences.addSetting({
-  id: "alwaysCheckDefault",
-  pref: "browser.shell.checkDefaultBrowser",
-  setup: emitChange => {
-    if (!DefaultBrowserHelper.canCheck) {
-      return undefined;
-    }
-    return DefaultBrowserHelper.pollForDefaultChanges(emitChange);
-  },
-  /**
-   * Show button for setting browser as default browser or information that
-   * browser is already the default browser.
-   */
-  visible: () => DefaultBrowserHelper.canCheck,
-  disabled: (_, setting) =>
-    !DefaultBrowserHelper.canCheck ||
-    setting.locked ||
-    DefaultBrowserHelper.isBrowserDefault,
-});
-
-Preferences.addSetting({
-  id: "isDefaultPane",
-  deps: ["alwaysCheckDefault"],
-  visible: () =>
-    DefaultBrowserHelper.canCheck &&
-    DefaultBrowserHelper.isBrowserDefault &&
-    Services.policies.isAllowed("setDefaultBrowser"),
-});
-
-Preferences.addSetting({
-  id: "isNotDefaultPane",
-  deps: ["alwaysCheckDefault"],
-  visible: () =>
-    DefaultBrowserHelper.canCheck &&
-    !DefaultBrowserHelper.isBrowserDefault &&
-    Services.policies.isAllowed("setDefaultBrowser"),
-  onUserClick: (e, { alwaysCheckDefault }) => {
-    if (!DefaultBrowserHelper.canCheck) {
-      return;
-    }
-    const setDefaultButton = /** @type {MozButton} */ (e.target);
-
-    if (!setDefaultButton) {
-      return;
-    }
-    if (setDefaultButton.disabled) {
-      return;
-    }
-
-    /**
-     * Disable the set default button, so that the user
-     * doesn't try to hit it again while browser is being set to default.
-     */
-    setDefaultButton.disabled = true;
-    alwaysCheckDefault.value = true;
-    DefaultBrowserHelper.setDefaultBrowser().finally(() => {
-      setDefaultButton.disabled = false;
-    });
-  },
-});
-
@@ -465,50 +400,0 @@ Preferences.addSetting({
-function createDefaultBrowserConfig({
-  includeIsDefaultPane = true,
-  inProgress = false,
-  hiddenFromSearch = false,
-} = {}) {
-  const isDefaultPane = {
-    id: "isDefaultPane",
-    l10nId: "is-default-browser-2",
-    control: "moz-promo",
-    controlAttrs: {
-      imagesrc: "chrome://global/skin/illustrations/kit-happy.svg",
-      imagedisplay: "cover",
-    },
-  };
-
-  const isNotDefaultPane = {
-    id: "isNotDefaultPane",
-    l10nId: "is-not-default-browser-2",
-    control: "moz-promo",
-    options: [
-      {
-        control: "moz-button",
-        l10nId: "set-as-my-default-browser-2",
-        id: "setDefaultButton",
-        slot: "actions",
-
-        controlAttrs: {
-          type: "primary",
-        },
-      },
-    ],
-    controlAttrs: {
-      imagesrc: "chrome://global/skin/illustrations/kit-concerned.svg",
-      imagedisplay: "cover",
-    },
-  };
-
-  const items = includeIsDefaultPane
-    ? [isDefaultPane, isNotDefaultPane]
-    : [isNotDefaultPane];
-
-  return {
-    l10nId: "home-default-browser-title",
-    headingLevel: 2,
-    items,
-    ...(inProgress && { inProgress }),
-    ...(hiddenFromSearch && { hiddenFromSearch }),
-  };
-}
-
@@ -568,4 +454,0 @@ function createStartupConfig(hidden = false) {
-      {
-        id: "alwaysCheckDefault",
-        l10nId: "always-check-default",
-      },
@@ -575,6 +457,5 @@ function createStartupConfig(hidden = false) {
 
 SettingGroupManager.registerGroups({
-  defaultBrowser: createDefaultBrowserConfig(),
   startup: createStartupConfig(
     Services.prefs.getBoolPref("browser.settings-redesign.enabled", false)
   ),
@@ -682,6 +563,5 @@ var gMainPane = {
     initSettingGroup("spellCheck");
     initSettingGroup("performance");
-    initSettingGroup("defaultBrowser");
     initSettingGroup("startup");
     initSettingGroup("importBrowserData");
     initSettingGroup("tabs");
